myst/src/s3/segment_download.rs

```diff
@@ -398,8 +398,12 @@ pub async fn start_download() -> Result<()> {
     Ok(())
 }
 
-pub async fn get_remote_shards(data_path: String, remote_store: Arc<RemoteStore>) -> Result<usize> {
-    let files = remote_store.list_files(data_path.clone()).await?;
+pub async fn get_remote_shards(data_path: &String, remote_store: Arc<RemoteStore>) -> Result<usize> {
+    let mut path = data_path.clone();
+    if ! path.ends_with("/") {
+        path.push_str("/");
```

Comment:
There is a util method which does this. However, now that I think about it, it maybe better to use a special Path object to take care of this. Will need more changes across the repo and can come later I guess.

Reply:
This will be contained in the `get_remote_shards` method. We probably won't have to do anything in other places.